### ISSUE TITLE
Plugin doesn't work with 10.7

### ISSUE THREAD

**Issue by Uatschitchun** [jellyfin/jellyfin-plugin-trakt#62]
Installed trakt v11 with jellyfin/jellyfin-unstable docker and it's not working.

When trying to authorize I get:
```
Alarm
An error occurred when trying to authorize device: 500 - Internal Server Error
```
And the logs show:
```
[15:39:53] [ERR] [51] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL POST /Trakt/Users/665c084a71f54088a04333e5e7754bc0/Authorize.
System.TypeLoadException: Could not load type 'MediaBrowser.Model.Serialization.IJsonSerializer' from assembly 'MediaBrowser.Model, Version=10.7.0.0, Culture=neutral, PublicKeyToken=null'.
   at System.Signature.GetSignature(Void* pCorSig, Int32 cCorSig, RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo methodHandle, RuntimeType declaringType)
   at System.Reflection.RuntimeConstructorInfo.get_Signature()
   at System.Reflection.RuntimeConstructorInfo.GetParametersNoCopy()
   at System.Reflection.RuntimeConstructorInfo.GetParameters()
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(ResultCache lifetime, Type serviceType, Type implementationType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, Type serviceType, CallSiteChain callSiteChain, Int32 slot)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateCallSite(Type serviceType, CallSiteChain callSiteChain)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.CreateServiceAccessor(Type serviceType)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.AspNetCore.Mvc.Controllers.ControllerFactoryProvider.<>c__DisplayClass5_0.<CreateControllerFactory>g__CreateController|0(ControllerContext controllerContext)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Jellyfin.Server.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager)
   at Jellyfin.Server.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager)
   at Jellyfin.Server.Middleware.IpBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Jellyfin.Server.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext)
   at Jellyfin.Server.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.ExceptionMiddleware.Invoke(HttpContext context)
[15:39:53] [WRN] [32] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://192.168.5.13:8096/Users/665c084a71f54088a04333e5e7754bc0/Items?Recursive=true&Limit=3&Filters=IsNotFolder&SortBy=DateCreated&SortOrder=Descending&Ids=68096cfa7400d9f99ec417c473e009bf%2C1870cf8aaaa171b6dfa16fa3cee90fd5%2Cc8be25f7a18c4bba5c270f61026d4d41%2C4a3b2e6768422bb8e17e0bd1ba336990%2C85cd12f743186166c6d656982eb810aa%2Ced79368adcfd7df4eaf53a3f63a741fd%2C455c9676c445ac39f014c66d3b0dcc31%2C93119ba9ca721b40e44a93f0750ff7e5%2C9bc83692148f351846f9f61152c833f6%2Cae3e8cc4225cad67fbd5a2fdc7dae5d3%2Cc5c5c97eb3ccfd28e07ad04e7a53aa1d%2C6ac45d2741ce192d5add6bdd9ca501de&MediaTypes=Audio%2CVideo&EnableTotalRecordCount=false to 192.168.5.215 in 0:00:00.5329628 with Status Code 200
```

A `docker logs -f jellyfin | grep -i trakt` gives:
```
[15:34:06] [INF] [27] Emby.Server.Implementations.Updates.InstallationManager: New plugin installed: Trakt 11.0.0.0
[15:34:06] [WRN] [27] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from http://192.168.5.13:8096/Packages/Installed/Trakt?AssemblyGuid=4fe3201e-d6ae-4f2e-8917-e12bda571281&version=11.0.0.0 to 192.168.5.215 in 0:00:00.8759662 with Status Code 204
[15:34:24] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded assembly Trakt, Version=11.0.0.0, Culture=neutral, PublicKeyToken=null from /config/plugins/Trakt_11.0.0.0/Trakt.dll
[15:34:39] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: Trakt 11.0.0.0
[15:34:39] [ERR] [1] Emby.Server.Implementations.ApplicationHost: Error creating Trakt.ScheduledTasks.SyncFromTraktTask
[15:34:39] [ERR] [1] Emby.Server.Implementations.ApplicationHost: Error creating Trakt.ScheduledTasks.SyncLibraryTask
[15:34:40] [ERR] [1] Emby.Server.Implementations.ApplicationHost: Error creating Trakt.ServerMediator
[15:36:36] [ERR] [60] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL POST /Trakt/Users/665c084a71f54088a04333e5e7754bc0/Authorize.
[15:36:59] [ERR] [11] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL POST /Trakt/Users/665c084a71f54088a04333e5e7754bc0/Authorize.
[15:39:53] [ERR] [51] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL POST /Trakt/Users/665c084a71f54088a04333e5e7754bc0/Authorize.
```

**Comment by EVOTk** [comment]
Hello,
i'm same error with plugin 11.2101.2400.2650

`
[01:53:35] [ERR] [34] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL POST /Trakt/Users/ec9df81564314128a23e45468884667a/Authorize.,
[01:52:57] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded assembly Trakt, Version=11.2101.2400.2650, Culture=neutral, PublicKeyToken=null from /config/data/plugins/Trakt_11.2101.2400.2650/Trakt.dll,
[01:53:10] [INF] [1] Emby.Server.Implementations.Plugins.PluginManager: Loaded plugin: Trakt 11.2101.2400.2650,
[01:53:10] [ERR] [1] Emby.Server.Implementations.ApplicationHost: Error creating Trakt.ScheduledTasks.SyncFromTraktTask,
[01:53:10] [ERR] [1] Emby.Server.Implementations.ApplicationHost: Error creating Trakt.ScheduledTasks.SyncLibraryTask,
[01:53:11] [ERR] [1] Emby.Server.Implementations.ApplicationHost: Error creating Trakt.ServerMediator
[01:53:35] [ERR] [34] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL POST /Trakt/Users/xxxxxxxxxxxxxxxxxxx/Authorize.`

**Comment by mattssn11** [comment]
Also having issues getting this to work on 10.7

**Comment by AlexDias95** [comment]
I have the same error with 10.7

**Comment by zshc** [comment]
Same error in 11.0

**Comment by haykkh** [comment]
Not sure why this was closed, still getting the same error on 10.7 with trakt 12.0
